### PR TITLE
Bug 37

### PR DESCRIPTION
**Pull request by nhenneaux**
Bug fix for https://github.com/SonarQubeCommunity/sonar-findbugs/issues/37.

### CODE REVIEW

File: src/test/java/org/sonar/plugins/findbugs/FindbugsSensorTest.java [hunk 1]
```diff
@@ -185,7 +182,7 @@ public void should_execute_findbugs_even_if_only_findsecbug() throws Exception {
   }
 
   @Test
-  public void should_execute_findbugs_but_not_find_violation() throws Exception {
+  public void should_not_execute_findbugs_if_no_active() throws Exception {
 
     BugInstance bugInstance = getBugInstance("THIS_RULE_DOES_NOT_EXIST", 107);
```
Comment (h3xstream) [inline src/test/java/org/sonar/plugins/findbugs/FindbugsSensorTest.java]: I think two different tests will be required. This test initially tested that bug instance with unknown id (for example a bug report that is not part of the Sonar descriptions or not supported).

File: src/main/java/org/sonar/plugins/findbugs/FindbugsSensor.java [hunk 2]
```diff
@@ -84,6 +84,9 @@ private boolean hasActiveFindSecBugsJspRules() {
 
   @Override
   public void execute(SensorContext context) {
+    if(!hasActiveFindbugsRules() && !hasActiveFbContribRules() && !hasActiveFindSecBugsRules() && !hasActiveFindSecBugsJspRules()){
+      return;
+    }
```
Comment (h3xstream) [inline src/main/java/org/sonar/plugins/findbugs/FindbugsSensor.java]: Perfect that is exactly what was needed.

Comment (nhenneaux) [inline src/main/java/org/sonar/plugins/findbugs/FindbugsSensor.java]: You are right, I have made an update.

**Comment by majidpal** [comment]
Hi - I am facing same issue so in which version this has fixed. I am using sonarqube 5.6 and plugins 3.3? Please help with details

**Comment by nhenneaux** [comment]
it has been fixed on plugin v3.4 (https://github.com/h3xstream/sonar-findbugs/issues/37)

**Comment by msaraf1** [comment]
Hi Team,

We are using the plugin V3.4 and sonarqube version 6.0.  getting the below error while running the sonar scan on the uncompiled code. sonar scanner v2.8

WARN: Findbugs needs sources to be compiled. Please build project before executing sonar or check the location of compiled classes to make it possible for Findbugs to analyse your project.
build	04-Jun-2018 18:27:01	INFO: ------------------------------------------------------------------------
build	04-Jun-2018 18:27:01	INFO: EXECUTION FAILURE
build	04-Jun-2018 18:27:01	INFO: ------------------------------------------------------------------------
build	04-Jun-2018 18:27:01	INFO: Total time: 4:58.293s
build	04-Jun-2018 18:27:01	INFO: Final Memory: 72M/12293M
error	04-Jun-2018 18:27:01	ERROR: Error during SonarQube Scanner execution
build	04-Jun-2018 18:27:01	INFO: ------------------------------------------------------------------------
error	04-Jun-2018 18:27:01	java.lang.IllegalStateException: Can not execute Findbugs

**Comment by majidpal** [comment]
You need to run sonar on compiled code.

**Comment by msaraf1** [comment]
Hi 
Just checking can we not use the sonar scanner to run the scan on the uncompiled code and without using the findbugs.

**Comment by majidpal** [comment]
Don't think so.

**Comment by msaraf1** [comment]
But i checked on the github repository that this is a bug and it has been fixed in the v3.4.4. Could you please check on the same.

**Pull request merged**